src/Reflection/ReflectionParameter.php

```diff
@@ -449,7 +466,10 @@ public function isDefaultValueConstant(): bool
      */
     public function getDefaultValueConstantName(): string
     {
-        $this->parseDefaultValueNode();
+        if ($this->defaultValueAst === null) {
```

Comment:
Are these AST caches necessary? I would suggest avoiding them, if they don't bring a massive advantage (the added complexity is noticeable)

Reply:
I remvoed the AST cache.